generate/generate.go

```diff
@@ -116,18 +116,19 @@ func (g *Generator) WriteFunctions() error {
 		}
 
 		var ue graphql.Type
+		var errorField string
 		for _, field := range ret.Fields {
-			if field.Name == "userErrors" {
+			if field.IsError() {
+				errorField = field.Name
 				ue, ok = g.schema.Type(field.TypeName())
 				if !ok {
 					return fmt.Errorf("could not find type for mutation user error: %s", field.TypeName())
 				}
-
 				break
 			}
 		}
 
-		mtm := newMutationTemplateModel(f, ret, g.schema.Mutation.Type, ue)
+		mtm := newMutationTemplateModel(f, ret, g.schema.Mutation.Type, ue, errorField)
```

Comment:
Seems like we should be able to change the params of this method if we make the graphql structs more capable. I could see it being something like

```golang
newMutationTemplateModel(g.schema, f)
```

And then inside we do the work to get what we need out of that mutation Field.

Reply:
Done, I also tried to move as much code as possible as struct funtion when it make sense. Let me know what you think. I tested it does generate the same code as previously :)